src/transformers/modeling_utils.py

```diff
@@ -3013,6 +3052,55 @@ def from_pretrained(
                 config._attn_implementation = kwarg_attn_imp
             model_kwargs = kwargs
 
+        # HQQ-logic adapted from https://github.com/mobiusml/hqq/blob/master/hqq/models/base.py#L175
+        quant_config = getattr(config, "quantization_config", None)
+        if check_if_hqq_quant_config(quant_config):
```

Comment:
Hmm indeed I think I need to think a bit how we could refactor this logic inside the quantizer. I would say let's first have a v1 without serialization / loading HQQ serialized weights from the Hub as it involves a lot of refactor, what do you think?